### PR TITLE
Add support for displaying "unranked PP" placeholder

### LINKED ISSUE #27027: Scores with unranked mods still show as calculating in-game
need to fix the in-game overlay to match the updated web logic (see https://github.com/ppy/osu-web/pull/10918)

### Discussed in https://github.com/ppy/osu/discussions/27026

<div type='discussions-op-text'>

<sup>Originally posted by **Flamiii** February  4, 2024</sup>
The display on the website for scores with unranked mods recently changed.

image

However, when looking at this score in the in-game webview, it still says that it is calculating when it isn't.

image
</div>

### PR DESCRIPTION
**Pull request by frenzibyte**
- [x] Depends on localisation for tooltip (needs `osu-resources` crowdin bump, cc @peppy)
- Closes https://github.com/ppy/osu/issues/27027

Not 100% sure about naming the "ranked" bool property as `Ranked` in `ScoreInfo`, but I'm not well aware about its intended usages, so I might be wrong if I said it's specifically for displaying "unranked PP" tooltip.

Also I've changed the icon of the "unprocessed PP" placeholder in line with https://github.com/ppy/osu-web/pull/10918/commits/5dd3db3be8cb7c05a0bef2d10c2bad4000fd4679.

### CODE REVIEW

File: osu.Game/Scoring/Drawables/UnrankedPerformancePointsPlaceholder.cs [hunk 1]
```diff
@@ -0,0 +1,26 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+#nullable disable
```
Comment (bdach) [inline osu.Game/Scoring/Drawables/UnrankedPerformancePointsPlaceholder.cs:4]:
```suggestion
```

File: osu.Game/Overlays/Profile/Sections/Ranks/DrawableProfileScore.cs [hunk 2]
```diff
@@ -216,7 +216,18 @@ private Drawable createDrawablePerformance()
             if (!Score.PP.HasValue)
             {
                 if (Score.Beatmap?.Status.GrantsPerformancePoints() == true)
+                {
+                    if (!Score.Ranked)
```
Comment (bdach) [inline osu.Game/Overlays/Profile/Sections/Ranks/DrawableProfileScore.cs:220]: Three nested ifs is the point where I call enough. Can you refactor this so that this doesn't happen? Rider tells me you can invert just ifs twice.

**Review by bdach — DISMISSED** [review]

File: osu.Game/Overlays/Profile/Sections/Ranks/DrawableProfileScore.cs [hunk 3]
```diff
@@ -213,11 +213,36 @@ private void load(RulesetStore rulesets)
...
+                            Origin = Anchor.BottomLeft,
+                            Font = OsuFont.GetFont(size: 12, weight: FontWeight.Bold),
+                            Text = "pp",
+                            Colour = colourProvider.Light3
+                        }
+                    }
+                };
+            }
 
+            if (Score.Beatmap?.Status.GrantsPerformancePoints() != true)
```
Comment (peppy) [inline osu.Game/Overlays/Profile/Sections/Ranks/DrawableProfileScore.cs:244]: Is there a reason this can't just be grouped in with the `UnrankedPerformanePointsPlaceholder` below? It seems like basically the same case.

File: osu.Game/Overlays/Profile/Sections/Ranks/DrawableProfileScore.cs
[the remark this answers is not in the document]
Comment (peppy) [reply osu.Game/Overlays/Profile/Sections/Ranks/DrawableProfileScore.cs:1]: I've commented on the lines responsible, but I think the goal is to have `Ranked` set to `false` in all cases where pp is not applicable. That said, there's probably no harm in being overbearing for cases we know satisfy this criteria, ie. non-ranked beatmaps.

File: osu.Game/Overlays/Profile/Sections/Ranks/DrawableProfileScore.cs (hunk 3 above)
Comment (bdach) [reply osu.Game/Overlays/Profile/Sections/Ranks/DrawableProfileScore.cs:244]: Well if we're being strict then t's not the same case. This handles stuff like scores on loved/graved beatmaps. Web uses a different tooltip for those ("Only ranked beatmaps award pp").

Comment (frenzibyte) [reply osu.Game/Overlays/Profile/Sections/Ranks/DrawableProfileScore.cs:244]: The duplication of the symbol looks really awful, I'll alter the code slightly.

Comment (bdach) [reply osu.Game/Overlays/Profile/Sections/Ranks/DrawableProfileScore.cs:244]: not sure that is necessary.

Comment (frenzibyte) [reply osu.Game/Overlays/Profile/Sections/Ranks/DrawableProfileScore.cs:245]: Might as well if it looks weird to someone already.

Comment (bdach) [reply osu.Game/Overlays/Profile/Sections/Ranks/DrawableProfileScore.cs:245]: Possibly. Relevant source:

https://github.com/ppy/osu-web/blob/64d540df6e8296e5090a9cc56a59189058585b67/resources/js/utils/beatmap-helper.ts#L114-L116
https://github.com/ppy/osu-web/blob/64d540df6e8296e5090a9cc56a59189058585b67/resources/js/profile-page/play-detail.tsx#L125

Comment (bdach) [reply osu.Game/Overlays/Profile/Sections/Ranks/DrawableProfileScore.cs:245]: Again, if we're being pedantic, then not quite because of https://github.com/ppy/osu-web/blob/64d540df6e8296e5090a9cc56a59189058585b67/resources/js/profile-page/play-detail.tsx#L126-L128

tl;dr loved maps should show a fontawesome heart instead of a dash.

**Review by peppy — APPROVED** [review]
Let's go with this for now.
